snippets/html.json

```diff
@@ -15,9 +15,45 @@
       "\t]",
       "\t}",
       "</script>",
-      "<script src=\"https://js.arcgis.com/${2|4.18,4.17,4.16,4.15,4.14|}\"></script>"
+      "<script src=\"https://js.arcgis.com/${2|4.19,4.18,4.17,4.16,4.15,4.14|}\"></script>"
     ],
     "description": "Add script and link tags for jsapi"
+  },
+  "Import ESM modules":{
+    "prefix": "importESM",
+    "body": [
+      "<link rel=\"stylesheet\" href=\"https://js.arcgis.com/${2|4.19,4.18|}/esri/themes/light/main.css\">",
+      "<style type=\"text/css\">",
+      "\thtml,",
+      "\tbody,",
+      "\t#viewDiv {",
+      "\t\tmargin: 0;",
+      "\t\tpadding: 0;",
+      "\t\twidth: 100%;",
+      "\t\theight: 100%;",
+      "\t}",
+      "</style>",
+      "<script type=\"module\">",
+      "\timport MapView from \"https://js.arcgis.com/${2|4.19,4.18|}/@arcgis/core/views/MapView.js\";", 
+      "\timport esriConfig from \"https://js.arcgis.com/${2|4.19,4.18|}/@arcgis/core/config.js\";", 
+      "\t", 
+      "\tesriConfig.apiKey = \"YOUR_FREE_API_KEY_FROM_DEVELOPERS.ARCGIS.COM\";", 
+      "\t", 
+      "\tconst view = new MapView({", 
```

Comment:
what is the difference between this and the CreateMap code snippet? https://github.com/Esri/arcgis-js-vscode-snippets/blob/master/snippets/javascript.json#L88

Back then we discussed that we don't want to import modules, we leave that up to the user's dev setup to be taken care of. There are many ways for a user to do that and we don't want to give them more work by having to delete our imports...

Reply:
You are right, it is pretty similar, the main difference was the import modules. Btw, since the [MapView is capable of autocast the map](https://developers.arcgis.com/javascript/latest/api-reference/esri-views-MapView.html#map) why do we still import the Map class and generate a variable just for the view?.  In my case, I try to avoid declaring variables whenever is possible, and this is usually one of those cases.

Anyway, I think in the end, I think it is very subjective, and **the type of snippets a developer would like to have/use will depend on the type of projects they use to work on** (client-side, server-side, 2D, 3D, ...), so I feel that the coolest thing would be to explain how to customize the extension with your own snippets, leave it clear a contributing guide (include in the contributing.md things like what you just explained here), and in the best scenario... have something like [Bootstrap customize](https://getbootstrap.com/docs/3.4/customize/) with community contributions,  allowing the developer to choose which ones likes to them 😄  (too much maybe? hehe)

Reply:
Thanks for contributing @hhkaos! I agree with @RalucaNicola that most (majority) of users are using local builds. My primary concern with the `importESM` snippet is that we don't want to give anyone the impression that they can use the ESM CDN in production applications. From a performance perspective, it's our most inefficient build and should only be used for prototyping and testing. Instead we are promoting npm installs of the ES modules as the best way to do local builds.

Reply:
Thanks Andy, I understand. So I'll remove it and try not to add more than one snippet per PR too ^_^

Reply:
Sounds good. One last recommendation, if you are able to wait 9 - 10 more days you can add this snippet along with the 4.20 release:

` "<script src=\"https://js.arcgis.com/${2|4.19,4.18,4.17,4.16,4.15,4.14|}\"></script>"`

Reply:
- you can add several snippets Raul, it's just that we want to stay away from using imports in them because there are several ways to import modules -> with a good setup, the editor should do that for you anyway

- as Andi said ESM CDN is more for small demos rather than production apps

- to answer your question about the map variable, I need the map to add layers or to find a certain layer. I guess I use a map variable to avoid having to go through the view to get it. But it's a matter of personal preference in the end.

Reply:
Thanks!

>  But it's a matter of personal preference in the end.

Yes you are right. In my case I think it adds extra complexity for new developers so in most cases I just try to use view.map.